python/pyspark/resource/taskresourcerequests.py

```diff
@@ -0,0 +1,52 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.resource.taskresourcerequest import TaskResourceRequest
+
+
+class TaskResourceRequests(object):
+
+    """
+    .. note:: Evolving
+
+    A set of task resource requests. This is used in conjuntion with the
+    ResourceProfileBuilder to programmatically specify the resources needed for
+    an RDD that will be applied at the stage level.
+    """
+
+    def __init__(self):
+        """Create a new TaskResourceRequests that wraps the underlying JVM object."""
+        from pyspark import SparkContext
+        self._javaTaskResourceRequests \
+            = SparkContext._jvm.org.apache.spark.resource.TaskResourceRequests()
```

Comment:
@tgravescs, I think we can lazily create this, e.g.)

```python
def __init__():
    self. _jTaskResourceRequests = None

@propery
def _javaTaskResourceRequests():
    if _jTaskResourceRequests is None:
        self._jTaskResourceRequests = \
            SparkContext._jvm.org.apache.spark.resource.TaskResourceRequests()
    return self._jTaskResourceRequests
```

In this way, users can create these resource instances safely regardless of the initialization of SparkContext. This can happen when you directly use PySpark library in a regular Python shell (or some other notebooks). PySpark types and UDF instances follow this way too.

Reply:
Ah, making it as a property seems not enough because it calls other JVM methods too.

Can we keep the given arguments in `self`, and construct a Java object at `withResources`?

Reply:
hmm, that seems like a lot of duplicate logic and memory usage. I'll take a closer look.

Reply:
so what the expected behavior when the jvm isn't available? ResourceProfile has a function to get the id. This has to be generated on the java side.  I assume if they don't have a jvm yet this just errors?  I think I can make everything else work and as long as they don't get the id it should be ok.  is there a policy we have followed for this?  I'll work on implementing this and see if I hit any other issues.

Reply:
There's no policy for it. It's just logically best to avoid JVM access whenever it's possible because it's flaky in general and expensive, and it makes users possible to create resource files before users launch `SparkContext`. Otherwise, users should always create `SparkContext` instances first. PySpark types and UDF don't have such restrictions.

Do you plan to expose `ResourceProfile.id` in PySpark side as an API? From my cursory look, it might not be very difficult to switch because the classes in Python side look simple; however, I could miss something. Is there any instance dependent on JVM side? If so, we'll have no choice but depending on JVM.

Reply:
yes I plan on exposing it, it got missed in the code posted here. I noticed it investigating this. I think that is the only case so I was just thinking of having it work unless they called into the resource profile.id function and then it would fail at that point, but wanted to see if there was a standard for handling that case.  Sounds like not so I'll just leave it like that